inc/Engine/Media/AboveTheFold/WarmUp/Controller.php

```diff
@@ -237,9 +238,15 @@ public function add_wpr_imagedimensions_query_arg( string $url ): string {
 	/**
 	 * Check if the current request is for mobile.
 	 *
+	 * @param bool $is_warm_up if check is for warm up or not.
+	 *
 	 * @return bool
 	 */
-	private function is_mobile(): bool {
+	private function is_mobile( bool $is_warm_up = false ): bool {
+		if ( $is_warm_up ) {
+			return true;
+		}
```

Comment:
While this solves the issue, the risk is that, if the user deactivate the separate mobile cache option, then we would not listen to this change and still always send desktop and mobile to the SaaS for the homepage.